### DISCUSSION TITLE
@app.route not show function in docs and redoc

### DISCUSSION THREAD

**Discussion by sugizo** [fastapi/fastapi#2663] [Questions] [answered]
**e.g. work well but not show in docs and redoc**
```
@app.route('/api/generate_token', methods = ['GET'] )
def generate_token(request):
    jwt_encode = jwt.encode(payload, secret, algorithm = algorithm)
    return responses.PlainTextResponse(jwt_encode)
```

**question**
why code above not show in docs and redoc ?
is it normal or there is lack on code above ?

thanks

**Comment by Mause** [comment, marked as answer]
Use `app.get` if you want it to appear in documentation

**Reply by krishnardt** [reply]
in Flask, you have @app.route functionality to define routes.

in FastAPI, you can directly write them as
@app.get()
@app.post() and
@app.put()

then they will appear in swagger docs or redocs.
